mautic/mautic
# Rebase dynamics UTC patch for M5.

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

Rebase of https://github.com/mautic/mautic/pull/10507 for M5

**Summary**

This statement Uses Non UTC dates and creates ZULU (UTC) dates from this in the parameters. (It explicitely cuts of the +02)
`printf('modifiedon ge %sZ', substr($params['start'], 0, '-6'))`
Zulu time: https://www.timeanddate.com/worldclock/timezone/zulu

The result is that in the countries with (eg +02) different timezones a set of records is fetched that is incomplete.

There should be something like:

```
$given = new DateTime($params['start']);
$given->setTimezone(new DateTimeZone("UTC"));
$oparams['$filter'] = sprintf('modifiedon ge %sZ', $given->format("Y-m-d\TH:i:s"));
```



#### Steps to test this PR:

See issue https://github.com/mautic/mautic/issues/10506


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.
